shared/util.go

```diff
@@ -7,18 +7,58 @@ package shared
 import (
 	"context"
 	"errors"
+	"flag"
 	"fmt"
 	"net/http"
+	"os"
+	"path/filepath"
 	"reflect"
 	"regexp"
 	"strings"
 
+	"cloud.google.com/go/compute/metadata"
 	mapset "github.com/deckarep/golang-set"
 	log "github.com/sirupsen/logrus"
 	"google.golang.org/appengine"
 	gaelog "google.golang.org/appengine/log"
 )
 
+// ProjectID is a flag for the Google Cloud Platform project ID, if different from ID detected from metadata service.
+var ProjectID = flag.String("project_id", "", "Google Cloud Platform project ID, if different from ID detected from metadata service")
+
+// InitProjectID changes the ProjectID flag to fall back to the ID fetched from metadata.
+func InitProjectID() {
```

Comment:
I don't think it's a good idea to move this into `shared`. The `metadata` package is only available on Flex, not on Standard.

Reply:
The package needs to be available in both api/search/cache/service and api/search/cache/index, and it's not very "cache" specific. Just because code doesn't work in both standard/flex doesn't disqualify it from /shared (e.g. datastore_ae & datastore_cloud)

Reply:
Reverted :)